wdl/ExpansionHunterDenovo.wdl

```diff
@@ -0,0 +1,332 @@
+      else
+        basename(sample_bam_or_cram_, ".cram")
+
+    call ComputeSTRProfile {
+      input:
+        filename = filename,
+        bam_or_cram = sample_bam_or_cram_,
+        bam_or_cram_index = bam_or_cram_index_,
+        reference_fasta = reference_fasta,
+        reference_fasta_index = reference_fasta_index,
+        min_anchor_mapq = min_anchor_mapq,
+        max_irr_mapq = max_irr_mapq,
+        ehdn_docker = ehdn_docker,
+        runtime_attr_override = runtime_attr_str_profile,
+        postfixes = postfixes
+    }
+  }
+
+  scatter (i in range(length(samples_status))) {
+    if (samples_status[i] == "case") {
+      File case_str_profile_json = ComputeSTRProfile.str_profile[i]
+    }
+    if (samples_status[i] == "control") {
+      File control_str_profile_json = ComputeSTRProfile.str_profile[i]
+    }
+  }
+
+  Array[File] cases_str_profile_json = select_all(case_str_profile_json)
+  Array[File] controls_str_profile_json = select_all(control_str_profile_json)
+
+  call STRAnalyze {
+    input:
+      cases = cases_str_profile_json,
+      controls = controls_str_profile_json,
+      reference_fasta = reference_fasta,
+      reference_fasta_index = reference_fasta_index,
+      postfixes = postfixes,
+      analysis_type = analysis_type,
+      str_comparison_type = str_comparison_type,
+      ehdn_docker = ehdn_docker,
+      runtime_attr_override = runtime_attr_analysis
+  }
+
+  output {
+    File multisample_profile = STRAnalyze.multisample_profile
+    Array[File] cases_locus = ComputeSTRProfile.locus
+    Array[File] cases_motif = ComputeSTRProfile.motif
+    Array[File] cases_str_profile = ComputeSTRProfile.str_profile
+    Array[File] analysis_results = STRAnalyze.results
+  }
+}
+
+task ComputeSTRProfile {
+  input {
+    String filename
+    File bam_or_cram
+    File bam_or_cram_index
+    File reference_fasta
+    File? reference_fasta_index
+    Int min_anchor_mapq
+    Int max_irr_mapq
+    String ehdn_docker
+    RuntimeAttr? runtime_attr_override
+    FilenamePostfixes postfixes
+  }
+
+  output {
+    File locus = "${filename}${postfixes.locus}"
+    File motif = "${filename}${postfixes.motif}"
+    File str_profile = "${filename}${postfixes.profile}"
+  }
+
+  # Defining this varialbe is requires since it
+  # will trigger localization of the file. The
+  # localized file is used by the `profile` subcommand
+  # of ExpansionHunterDenovo.
+  File reference_fasta_index_ = select_first([
+    reference_fasta_index,
+    reference_fasta + ".fai"])
+
+  command <<<
+    set -euxo pipefail
+
+    ExpansionHunterDenovo profile \
+    --reads ~{bam_or_cram} \
+    --reference ~{reference_fasta} \
+    --output-prefix ~{filename} \
+    --min-anchor-mapq ~{min_anchor_mapq} \
+    --max-irr-mapq ~{max_irr_mapq}
+
+  >>>
+
+  RuntimeAttr runtime_attr_str_profile_default = object {
+    cpu_cores: 1,
+    mem_gb: 4,
+    disk_gb: 10,
+    boot_disk_gb: 10,
+    preemptible_tries: 0,
+    max_retries: 1
```

Comment:
Before merging, I would at least try to make disk_gb scale with input size, ie: `disk_gb: 10 + size([reference_fasta, bam_or_cram]),`

We can put off memory optimizations until you have done tests with real BAMs.

Also how long does this task typically take? Rule of thumb is if it takes less than 2 hours then set `preemptible_tries: 3`, otherwise try to parallelize the task with a `scatter`. If the latter is not possible we use `preemptible_tries: 0` but the cost is 5x higher. 

Same applies to `STRanalyze` below.

Reply:
Good point, thanks! I updated accordingly.